amp-viewer.js

```diff
@@ -8,116 +8,123 @@
  * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
  */
 
-(function(win) {
-  'use strict';
+'use strict'; {
 
-  document.registerElement('amp-viewer', {
-    prototype: {
-      __proto__: HTMLElement.prototype,
+  customElements.define('amp-viewer', class extends HTMLElement {
 
-      createdCallback: function() {
-        this._amp = null;
-        this._src = '';
-        this._host = null;
+    constructor() {
+      super();
 
-        if (!win.AMP_SHADOW) {
-          win.AMP_SHADOW = true;
-          this._installScript('https://cdn.ampproject.org/shadow-v0.js');
-        }
+      this._amp = null;
+      this._src = '';
+      this._host = null;
+    }
 
-        if (this.hasAttribute('src')) {
-          this.src = this.getAttribute('src');
-        }
-      },
+    connectedCallback() {
 
-      set src(src) {
-        if (this._src === src) {
-          return;
-        }
+      if (!window.AMP_SHADOW) {
+        window.AMP_SHADOW = true;
+        this._installScript(
+          'https://cdn.ampproject.org/shadow-v0.js');
+      }
 
-        this._src = src;
-        this._clear();
-        if (!src) {
-          this.removeAttribute('src');
-        } else {
-          this.setAttribute('src', src);
-          this._loadDocument(src);
-        }
-      },
+      if (this.hasAttribute('src')) {
```

Comment:
Custom Elements v1 has the [`observedAttributes`](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Custom_Elements#Observed_attributes) getter that allows you to tell which attributes you want to observe.
You should delete this check and just tell the browser you're interested on `src` attribute changes only, with:
```js
static get observedAttributes() {return ['src']; }